executor/trace.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/opentracing/basictracer-go"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/tracing"
+	"golang.org/x/net/context"
+)
+
+var traceColumns = append([]*types.FieldType{})
+
+// TraceExec represents a root executor of trace query.
+type TraceExec struct {
+	baseExecutor
+	// CollectedSpans collects all span during execution. Span is appended via
+	// callback method which passes into tracer implementation.
+	CollectedSpans []basictracer.RawSpan
+	// exhausted being true means there is no more result.
+	exhausted bool
```

Comment:
why we need this attribute?

Reply:
It indicates when `TraceExec` is finished.